MediaBrowser.Model/Cryptography/PasswordHash.cs

```diff
@@ -0,0 +1,110 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace MediaBrowser.Model.Cryptography
+{
+    public class PasswordHash
+    {
+        //Defined from this hash storage spec
+        //https://github.com/P-H-C/phc-string-format/blob/master/phc-sf-spec.md
+        //$<id>[$<param>=<value>(,<param>=<value>)*][$<salt>[$<hash>]]
+
+        public string Id;
+        public Dictionary<string, string> Parameters = new Dictionary<string, string>();
+        public string Salt;
+        public byte[] SaltBytes;
+        public string Hash;
+        public byte[] HashBytes;
+        public PasswordHash(string StorageString)
+        {
+            string[] a = StorageString.Split('$');

```

Comment:
Now I don't understand the reasons of storing salt right next to the hash, what use it has?.. I always thought salt should be stored elsewhere, so if a DB with user hashes is stolen it's useless without salt...

Reply:
Agreed. This is probably one of the few "user settings" that belongs in an XML file. And whatever file it ends up in, it should be the only thing in it as well imo.

Makes it less likely to be leaked by association that way.

Reply:
https://www.owasp.org/index.php/Password_Storage_Cheat_Sheet We've also talked about this in chat a fair bit, but basically storing the salt elsewhere offers little to no benefit vs the benefit from implementing a consistent standard of how to store passwords in a db

Reply:
I think this sums it up pretty well: https://security.stackexchange.com/a/183086

Reply:
Yeah, that owasp link got me enlightened.